searx/plugins/tracker_url_remover.py

```diff
@@ -2,43 +2,58 @@
 # pylint: disable=missing-module-docstring
 
 from __future__ import annotations
+import typing
 
 import re
 from urllib.parse import urlunparse, parse_qsl, urlencode
 
 from flask_babel import gettext
 
+from searx.plugins import Plugin, PluginInfo
+
+if typing.TYPE_CHECKING:
+    from searx.search import SearchWithPlugins
+    from searx.extended_types import SXNG_Request
+    from searx.result_types import Result
+    from searx.plugins import PluginCfg
+
 regexes = {
     re.compile(r'utm_[^&]+'),
     re.compile(r'(wkey|wemail)[^&]*'),
     re.compile(r'(_hsenc|_hsmi|hsCtaTracking|__hssc|__hstc|__hsfp)[^&]*'),
     re.compile(r'&$'),
 }
 
-name = gettext('Tracker URL remover')
-description = gettext('Remove trackers arguments from the returned URL')
-default_on = True
-preference_section = 'privacy'
-
 
-def on_result(_request, _search, result) -> bool:
+class SXNGPlugin(Plugin):
+    """Remove trackers arguments from the returned URL"""
+
+    id = "tracker_url_remover"
+
+    def __init__(self, plg_cfg: "PluginCfg") -> None:
+        super().__init__(plg_cfg)
+        self.info = PluginInfo(
+            id=self.id,
+            name=gettext("Tracker URL remover"),
+            description=gettext("Remove trackers arguments from the returned URL"),
+            preference_section="privacy",
+        )
+
+    def on_result(
+        self, request: "SXNG_Request", search: "SearchWithPlugins", result: Result
+    ) -> bool:  # pylint: disable=unused-argument
+        if not result.parsed_url:
+            return True
+
+        parsed_query = parse_qsl(result.parsed_url.query)
+        changes = 0
+        for i, (param_name, _) in enumerate(list(parsed_query)):
+            for reg in regexes:
+                if reg.match(param_name):
+                    parsed_query.pop(i - changes)
+                    changes += 1
```

Comment:
```suggestion
                    parsed_query.remove(param_name)
```
That's really terrifying code, although that's unrelated to this PR this plugin should probably be reworked (or removed, as discussed in https://github.com/searxng/searxng/issues/4472.

Reply:
I have adopted your suggestion ... however, the code needs to be revised ;-)